### DISCUSSION TITLE
successful confirmVerification doesn't set verified to true

### DISCUSSION THREAD

**Discussion by dschissler** [pocketbase/pocketbase#2401] [Q&A]
I think that the results of confirmVerification should set the verified to true before returning it.  It seems not to be the case.

**Comment by ganigeorgiev** [comment]
We have [unit tests](https://github.com/pocketbase/pocketbase/blob/master/forms/record_verification_confirm_test.go) for the confirm verification.

If you mean in the SDK, then Yes, it won't update the current model in the `pb.authStore` (if there is any).

The sucess API response is 204 and you have to call manually `authRefresh()` if you want to fetch the latest user data.

We can eventually manually check if the verification token payload matches the existing authenticated user (if any) and update the `verified` field, but then we'll not have the changes from the other fields like the `updated` timestamp.

**Reply by ganigeorgiev** [reply]
We could add an auto `authRefresh()` call on success confirm verification (and maybe the other confirm actions) if the jwt token payload matches the current authenticated user. But this will work only if you have custom confirm page (and the token is for the current user). 
I'll think on it more clearly tomorrow.

**Reply by dschissler** [reply]
@ganigeorgiev Yes I'm able to easily get around this using `authRefresh` afterwards, but it was just another development hiccup.

**Reply by ganigeorgiev** [reply]
I understand that it may not be expected but all non-auth email confirm actions return empty body with 204 (you can check some of the reasoning for this in https://github.com/pocketbase/pocketbase/discussions/1036).

**Reply by dschissler** [reply]
I think that the docs could use some major upgrades around this (after you've fully transitioned to the new options object format).

I'm thinking that each method should take a single object after `email`, `password` or `token`, and that which specific extras that it takes is wrapped up in the object inheritance.  So no more empty `{}` as third parameter to get to 4 parameter query params.  So some methods would take up to like 5 options on the options and others just 2.

Then more and better documentation for all of that.  So if `confirmVerification`  doesn't refresh auth token then show that in the example the next line (with comment).
